scipy/stats/_morestats.py

```diff
@@ -2885,6 +2885,83 @@ def fligner(*samples, center='median', proportiontocut=0.05):
     return FlignerResult(Xsq, pval)
 
 
+@_axis_nan_policy_factory(lambda x1: (x1,), n_samples=4, n_outputs=1)
+def _mood_inner_lc(xy, x, diffs, srted, n, m, N) -> float:
+    # obtain the unique values and the counts of each.
+    # "a_i, + b_i, = t_i, for j = 1, ... k", where `k` is the number of unique
+    # classes, and "[t]he number of values associated with the x's and y's in
+    # the jth class will be denoted by a_i, and b_i respectively."
+    # (Mielke, 312)
+    # Prepend `diffs` with a non-zero to indicate that the first element should
+    # be marked as not matching what preceded it.
+    diffs_prep = np.concatenate(([1], diffs))
+    # Unique elements are where the was a difference between elements in the
+    # sorted array
+    uniques = srted[diffs_prep != 0]
+    # The count of each element is the bin size for each set of consecutive
+    # differences where the difference is zero. Replace nonzero differences
+    # with 1 and then use the cumulative sum to count the indices.
+    t = np.bincount(np.cumsum(diffs_prep != 0))[1:]
+    k = len(uniques)
+    js = list(np.arange(1, k + 1))
```

Comment:
```suggestion
    js = list(range(1, k + 1))
```

nit: small simplification